volttron/platform/agent/base_historian.py

```diff
@@ -1580,7 +1575,14 @@ def get_outstanding_to_publish(self, size_limit):
             headers = {} if row[5] is None else loads(row[5])
             meta = self._meta_data[(source, topic_id)].copy()
             topic = self._backup_cache[topic_id]
+
             if (topic, timestamp) in dedup:
+                _log.debug(f"Found duplicate; removing record from cache: {row}")
```

Comment:
These should be batched so that they can be deleted all through one transaction.

Reply:
Good point. Will change.